l10n_br_account/models/account_invoice.py

```diff
@@ -136,6 +134,8 @@ def _shadowed_fields(self):
     def _prepare_shadowed_fields_dict(self, default=False):
         self.ensure_one()
         vals = self._convert_to_write(self.read(self._shadowed_fields())[0])
+        vals["partner_id"] = self.partner_id.commercial_partner_id.id
+        vals["partner_shipping_id"] = self.partner_shipping_id.commercial_partner_id.id
```

Comment:
Neto fiquei apenas com duvida em relação ao partner_shipping_id ele deveria mesmo assumir o comercial_partner_id ? no caso o endereço assumido deveria ser o partner_shipping e o cnpj e afins do comercial partner talvez. oq acha ?

Reply:
@marcelsavegnago pois é essa questão do endereço de entrega tem que ser estudado melhor, realmente não tá legal, vou rever isso

Reply:
para dar a voces um pouco de contexto do historico desses problemas entre empresa e contato (no inicio o partner_id era a empresa, e a Odoo SA mudou isso de forma bem aproximativa na v7 e eu vi eles ainda corrigir merdas disso na v16 recentemente, ou seja 10 anos depois meio que como eu previ na época...) https://www.slideshare.net/RaphalValyi/openerp-contact-id-solution-to-the-contacts-issue-in-v7
:-D

Reply:
mais: essa merda com os endereços foi meio que a gota que fez a gente criar a OCA em 2013 (isso e vazamentos internos que avisavam que a Odoo SA tinha agenda secreta e que envolvia ferrar os parceiros históricos).